Realm/Realm/Realm.cs

```diff
@@ -1002,9 +1011,9 @@ public async Task<T> WriteAsync<T>(Func<T> function)
                 return Write(function);
             }
 
-            using var transaction = await BeginWriteAsync();
+            using var transaction = await BeginWriteAsync(cancellationToken);
             var result = function();
-            await transaction.CommitAsync();
+            await transaction.CommitAsync(cancellationToken);
```

Comment:
We should not be passing the cancellation token to `CommitAsync` here - the reason is that it implies that the transaction will be cancelled, when this is not really the case. Consider the following code:

```csharp
try
{
    var cts = ...;
    await realm.WriteAsync(() => { ... }, cts);
}
catch (TaskCancelledException)
{
    // Here the user expects the transaction has been cancelled.
    // If we only canceled `CommitAsync`, the write will have
    // succeeded and will eventually be committed.
}
```

Reply:
I've had an exchange with @papafe exactly on the usefulness of the CancellationToken on the `CommitAsync` and exactly the unknown that could come from not knowing what the `WriteAsync` actually didn't do. I also wrote my initial thoughts on it in the description of this PR.
@papafe mentioned that's quite common for async calls with CancellationToken (e.g. http request) not to know what's actually been done. It made sense to me there that the user would check after the operation to know what was written and what not.
I suppose that your point is to keep things simple, which is a pretty good point. Let's see if @papafe has anything to add. Otherwise I'll go ahead with removing the ct from the `CommitAsync` call.

Reply:
I think it's fine to have a cancellation token for `CommitAsync` (i.e. if the user wants to manually pass a token there, that's fine). My argument is only for `WriteAsync` - i.e. there we only pass the cancellation token to `BeginWriteAsync` and not to `CommitAsync`. I'm thinking something like:

```csharp
using var transaction = await BeginWriteAsync(cancellationToken);
var result = function();

// Last chance to rollback the transaction - we manually check for cancellation
if (cancellationToken.IsCancellationRequested)
{
    throw new TaskCanceledException();
}

// No cancellation token here because we can't really cancel the commit
await transaction.CommitAsync();
```